Sources/SmithyTestUtil/ProtocolTestClient.swift

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+import protocol SmithyHTTPAPI.HTTPClient
+import class SmithyHTTPAPI.SdkHttpRequest
+import class SmithyHTTPAPI.HttpResponse
+import ClientRuntime
+
+public class ProtocolTestClient {
+    public init() {}
+}
+
+public enum TestCheckError: Error {
+    case actual(SdkHttpRequest)
+}
+
+extension ProtocolTestClient: HTTPClient {
+    public func send(request: SdkHttpRequest) async throws -> HttpResponse {
+        throw TestCheckError.actual(request)
```

Comment:
throwing an error with the request will allow tests to do checks against the result